panel/param.py

```diff
@@ -817,13 +821,17 @@ def _validate_object(self):
             "class or global state in an application's namespace."
         )
 
+    @classmethod
+    def applies(cls, obj: Any) -> float | bool | None:
```

Comment:
```suggestion
    def applies(cls, obj: Any) -> bool:
```

Reply:
I'm actually not entirely sure what the correct thing to do here is. Reflect on the actual method or the possibility of changes in inherited classes.

Reply:
Same, was never quite clear to me.

Reply:
Okay, I think covariant typing is generally what you should do so in principle it would be fine to narrow the type here. However I actually do not want to box things in for these cases. Subclasses should be allowed to override `applies` and give it a different  return type, e.g. you might make a very specific subclass of `ParamRef` that only supports a certain reference type but returns a float value in `applies` so it actually has a display precedence.